### PR TITLE
Add padding top if not set already

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
If the user has not set a padding top add it. Prevent decoration on first row getting cut off. 

I found no way of actually setting a default value. Now we check if the user has changed the value and if not set it for them.

fixes: https://github.com/pokey/cursorless-talon/issues/93

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -8,11 +8,14 @@ import { TestCaseRecorder } from "./testUtil/TestCaseRecorder";
 import { getCommandServerApi, getParseTreeApi } from "./util/getExtensionApi";
 import isTesting from "./testUtil/isTesting";
 import CommandRunner from "./core/commandRunner/CommandRunner";
+import updateMissingSetting from "./util/updateMissingSetting";
 
 export async function activate(context: vscode.ExtensionContext) {
   const { getNodeAtLocation } = await getParseTreeApi();
   const commandServerApi = await getCommandServerApi();
 
+  updateMissingSetting("editor.padding.top", 5);
```
Comment (pokey) [inline src/extension.ts:17]: Hesitant to add to this file, as we're currently trying to remove as much as we can. Maybe this belongs in init function of `Decorations`?

**Review by pokey — COMMENTED** [review]
LGTM tho I wanna quick test this one locally to see if that 5px is bothersome at all

File: src/extension.ts (hunk 1 above)
Comment (AndreasArvidsson) [reply src/extension.ts:17]: Even though this helps with decoration problem something as core as updating the user settings I still believe belongs in this file. In the case we have to add another setting it's nice to have all the changes we make to the user setting in one place.

**Comment by AndreasArvidsson** [comment]
@pokey On windows all the snippet tests are now failing for some reason. And only on windows of course

File: src/extension.ts (hunk 1 above)
Comment (pokey) [reply src/extension.ts:17]: Ok, I don't feel super strongly so let's leave it here

**Comment by pokey** [comment]
> @pokey On windows all the snippet tests are now failing for some reason. And only on windows of course

hmm yeah I noticed that on one of my branches too.  This must have just started, because I was able to merge a branch in earlier today.  Strange.  Does it work locally on Windows?

**Comment by pokey** [comment]
Do you think we could use [this new feature](https://code.visualstudio.com/updates/v1_63#_configuration-defaults-overrides) from 1.63.0 to do this one more simply?  Just saw it in the release notes.  We'd need to set min vscode version to 1.63.0, which is maybe a bit aggressive 4 days after it was released?

**Comment by AndreasArvidsson** [comment]
I do think that solution is better but wouldn't that require every user have the latest version of vscode? I don't know if everyone always running automatic updates.
At least think which should roll out one solution. I will let you choose which one.

**Comment by pokey** [comment]
> I do think that solution is better but wouldn't that require every user have the latest version of vscode? I don't know if everyone always running automatic updates. At least think which should roll out one solution. I will let you choose which one.

As long as calling `paddingConfig.update` is not persistent, I'd lean towards avoiding any new APIs at least for another week or two.  Alternately, we could leave this PR out of the 0.24.0 release.  But if it is non-persistent, then I think I'd be fine shipping this one, esp as it's a fairly modest code change.  I'm not seeing a big downside

**Comment by AndreasArvidsson** [comment]
The implementation in this pr is permanent. We actually updating the users settings.json

**Comment by pokey** [comment]
Ah ok right. In that case I'd lean towards reworking this one to use the new api, but waiting to merge until after 0.24.4 is released. Make sense?

**Comment by AndreasArvidsson** [comment]
I think so

**Comment by pokey** [comment]
> I think so

Shall we close this one then?  I don't think there will be any code reused in the new implementation

**Pull request closed without merging**